entity-framework/core/miscellaneous/events/interception.md

```diff
@@ -0,0 +1,750 @@
+
+Interceptors are registered per DbContext instance when the context is configured. Use a [diagnostic listener](xref:core/miscellaneous/events/diagnostics) to get the same information but for all DbContext instances in the process.
+
+## Registering interceptors
+
+Interceptors are registered using <xref:Microsoft.EntityFrameworkCore.DbContextOptionsBuilder.AddInterceptors%2A> when [configuring a DbContext instance](xref:core/miscellaneous/configuring-dbcontext). This is commonly done in an override of <xref:Microsoft.EntityFrameworkCore.DbContext.OnConfiguring%2A?displayProperty=nameWithType>. For example:
+
+<!--
+public class ExampleContext : BlogsContext
+{
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) 
+        => optionsBuilder.AddInterceptors(new TaggedQueryCommandInterceptor());
+}
+-->
+[!code-csharp[RegisterInterceptor](../../../../samples/core/Miscellaneous/CommandInterception/Program.cs?name=RegisterInterceptor)]
+
+Alternately, `AddInterceptors` can be called as part of <xref:Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.AddDbContext%2A> or when creating a <xref:Microsoft.EntityFrameworkCore.DbContextOptions> instance to pass to the DbContext constructor.
+
+> [!TIP]
+> OnConfiguring is still called when AddDbContext is used or a DbContextOptions instance is passed to the DbContext constructor. This makes it the ideal place to apply context configuration regardless of how the DbContext is constructed.
+
+Interceptors are often stateless, which means that a single interceptor instance can be used for all DbContext instances. For example:
+
+<!--
+public class TaggedQueryCommandInterceptorContext : BlogsContext
+{
+    private static readonly TaggedQueryCommandInterceptor _interceptor 
+        = new TaggedQueryCommandInterceptor();
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) 
+        => optionsBuilder.AddInterceptors(_interceptor);
+}
+-->
+[!code-csharp[RegisterStatelessInterceptor](../../../../samples/core/Miscellaneous/CommandInterception/Program.cs?name=RegisterStatelessInterceptor)]
+
+Every interceptor instance must implement one or more interface derived from <xref:Microsoft.EntityFrameworkCore.Diagnostics.IInterceptor>. Each instance should only be registered once even if it implements multiple interception interfaces; EF Core will route events for each interface as appropriate.
+
+## Database interception
+
+> [!NOTE]
+> Database interception was added in EF Core 3.0 and is only available for relational database providers.
+> Savepoint support was added in EF Core 5.0.
+
+Low-level database interception is split into the three interfaces shown in the following table.
```

Comment:
```suggestion
Low-level database interception is split into the three interfaces shown in the following table:
```

Reply:
Should this really be a colon when referring to a table?